src/btllib/util.cpp

```diff
@@ -157,6 +157,28 @@ get_basename(const std::string& path)
   return path.substr(p + 1);
 }
 
+double
+calc_phred_avg(const std::string& qual, const size_t start_pos, size_t len)
+{
+  if (len == 0) {
+    len = qual.size() - start_pos;
+  }
+
+  if (start_pos + len > qual.size()) {
+    throw std::invalid_argument(
```

Comment:
Should we consider using btllib's `log_error` here instead? (ex. https://github.com/bcgsc/btllib/blob/master/src/btllib/seq_writer.cpp#L40-#L42)